src/bundle/Resources/public/js/scripts/core/multilevel.popup.menu.js

```diff
@@ -0,0 +1,385 @@
+            );
+            branchElement.addEventListener(
+                'mouseenter',
+                () => {
+                    this.hoveredBranches.add(branchElement);
+                    this.updateBranchAndParentBranchesOpenState(branchElement);
+                },
+                false,
+            );
+            branchElement.addEventListener(
+                'mouseleave',
+                () => {
+                    setTimeout(() => {
+                        this.hoveredBranches.delete(branchElement);
+                        this.updateBranchAndParentBranchesOpenState(branchElement);
+                    }, 50);
+                },
+                false,
+            );
+
+            processBranchAfter(branchElement);
+            branchItems.forEach((itemElement) => processBranchItemAfter(itemElement));
+        }
+
+        updateBranchAndParentBranchesOpenState(branchElement) {
+            const isTopBranch = !(branchElement?.parentBranchElement ?? null);
+
+            if (isTopBranch) {
+                return;
+            }
+
+            this.updateBranchOpenState(branchElement);
+            this.updateBranchAndParentBranchesOpenState(branchElement.parentBranchElement);
+        }
+
+        updateBranchOpenState(branchElement) {
+            const isSubbranchOpened = (otherBranchElement) => {
+                return (
+                    otherBranchElement &&
+                    (branchElement === otherBranchElement || isSubbranchOpened(otherBranchElement.parentBranchElement))
+                );
+            };
+            const isBranchOrAnySubbranchHovered = [...this.hoveredItemsBranches, ...this.hoveredBranches].some(isSubbranchOpened);
+
+            if (isBranchOrAnySubbranchHovered) {
+                this.openBranch(branchElement);
+            } else {
+                this.closeWithSubbranches(branchElement);
+            }
+        }
+
+        async handleItemWithSubitemsClick(event) {
+            const itemWithSubitems = event.currentTarget;
+            const { branchElement } = itemWithSubitems;
+            const isExpanded = !branchElement.classList.contains('ibexa-popup-menu--hidden');
+            const shouldBeExpanded = !isExpanded;
+
+            if (shouldBeExpanded) {
+                this.openBranch(branchElement);
+            } else {
+                this.closeWithSubbranches(branchElement, shouldBeExpanded);
+            }
+        }
+
+        async openBranch(branchElement) {
+            this.toggleBranch(branchElement);
+            await branchElement.popperInstance.update();
+        }
+
+        closeBranch(branchElement) {
+            this.toggleBranch(branchElement, false);
+        }
+
+        toggleBranch(branchElement, shouldBeExpanded = true) {
+            const topBranch = this.triggerElement.branchElement;
+
+            branchElement.classList.toggle('ibexa-popup-menu--hidden', !shouldBeExpanded);
+
+            if (branchElement === topBranch) {
```

Comment:
does it work correctly? if you compare two DOM elements shouldn't you use `isEqualNode` method?

Reply:
it should work

Reply:
> There is no need to use isSameNode(); instead use the === strict equality operator.
ref. https://developer.mozilla.org/en-US/docs/Web/API/Node/isSameNode